hw/bsp/dialog_da1469x-dk-pro/otp_tool.py

```diff
@@ -0,0 +1,538 @@
+def otp_read_config(uart, outfile):
+    try:
+        ser = serial.Serial(port=uart, baudrate=115200, timeout=1,
+                            bytesize=8, stopbits=serial.STOPBITS_ONE)
+    except serial.SerialException:
+        raise SystemExit("Failed to open serial port")
+
+    # length is unused, so set to 0
+    cmd = cmd_append_value(0xaa55aa55, Cmd.OTP_READ_CONFIG, 0)
+    data = struct.pack('III', *cmd)
+
+    try:
+        ser.write(data)
+    except serial.SerialException:
+        raise SystemExit("Failed to write to %s" % uart)
+
+    data = ser.read(16)
+    if len(data) != 16:
+        raise SystemExit("Failed to receive response, exiting")
+
+    response = cmd_response._make(struct.unpack_from('IIII', data))
+    if response.status == 0:
+        data = ser.read(response.length)
+        if len(data) != response.length:
+            raise SystemExit("Failed to receive data, exiting")
+
+        print("Successfully read configuration script, writing to file: " + outfile)
+    try:
+        with open(outfile, "wb") as f:
+            f.write(data)
+    except IOError:
+        raise SystemExit("Failed to write config data to file")
+
+
+@click.argument('outfile')
+@click.option('-a', '--address', type=str, required=True,
+              help='flash address, hexadecimal')
+@click.option('-l', '--length', type=int, required=True, help='length to read')
+@click.option('-b', '--block-size', type=int, default=1024, help='block size')
+@click.option('-u', '--uart', required=True, help='uart port')
+@click.command(help='Read from flash')
+def flash_read(uart, length, outfile, address, block_size):
+    try:
+        ser = serial.Serial(port=uart, baudrate=115200, timeout=1,
+                            bytesize=8, stopbits=serial.STOPBITS_ONE)
+    except serial.SerialException:
+        raise SystemExit("Failed to open serial port")
+
+    try:
+        f = open(outfile, "wb")
+    except IOError:
+        raise SystemExit("Failed to open output file")
+
+    bytes_left = length
+    offset = int(address, 16)
+    while bytes_left > 0:
+        if bytes_left > block_size:
+            trans_length = block_size
+        else:
+            trans_length = bytes_left
+
+        cmd = cmd_flash_op(0xaa55aa55, Cmd.FLASH_READ, offset, trans_length)
+        data = struct.pack('IIII', *cmd)
+
+        try:
+            ser.write(data)
+        except serial.SerialException:
+            raise SystemExit("Failed to write cmd to %s" % uart)
+
+        data = ser.read(16)
+        if len(data) != 16:
+            raise SystemExit("Failed to receive response, exiting")
+
+        response = cmd_response._make(struct.unpack_from('IIII', data))
+        if response.status == 0:
+            data = ser.read(response.length)
+            if len(data) != response.length:
+                raise SystemExit("Failed to receive response, exiting")
+            f.write(data)
+
+        else:
+            SystemExit("Error in read response, exiting")
```

Comment:
Isn't this suppose to `raise` the exception? Also the `break` in the next line can be removed.

Reply:
done